### PR TITLE
Upgrade to JNA 5.5.0

### PR DESCRIPTION
**Pull request by jasontedor**
This commit bumps our JNA dependency from 4.5.1 to 5.5.0, so that we are now on the latest maintained line, and pick up a large collection of bug fixes that have accumulated.

### CODE REVIEW

**Review by pugnascotia — APPROVED** [review]

**Comment by jasontedor** [comment]
@elasticmachine update branch

File: buildSrc/build.gradle [hunk 1]
```diff
@@ -117,7 +117,7 @@ dependencies {
   compile 'com.netflix.nebula:nebula-publishing-plugin:4.4.4'
   compile 'com.netflix.nebula:gradle-info-plugin:7.1.3'
   compile 'org.apache.rat:apache-rat:0.11'
-  compile "org.elasticsearch:jna:4.5.1"
+  compile "org.elasticsearch:jna:5.5.0"
```
Comment (mark-vieira) [inline buildSrc/build.gradle:120]: Why don't we use the `version.properties` version here? We load this file even for the `buildSrc` build so we can reference it here, as we de below for things like checkstyle and randomized runner. That would save us another place to bump this version in the future.

**Review by rjernst — APPROVED** [review]
LGTM

**Comment by jasontedor** [comment]
@elasticmachine update branch

**Comment by jasontedor** [comment]
@elasticmachine update branch

**Comment by jasontedor** [comment]
@elasticmachine update branch

File: buildSrc/build.gradle (hunk 1 above)
Comment (jasontedor) [reply buildSrc/build.gradle:120]: I don't know why, can you pick this up in a follow-up?

**Pull request merged**